modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ERegex.java

```diff
@@ -68,8 +68,9 @@ void analyze(Locals locals) {
 
         try {
             Pattern.compile(pattern, flags);
-        } catch (PatternSyntaxException exception) {
-            throw createError(exception);
+        } catch (PatternSyntaxException e) {
+            throw new Location(location.getSourceName(), location.getOffset() + 1 + e.getIndex()).createError(
```

Comment:
Is it not possible to fix this value where the Location object is originally generated?